internal/plugins/golang/v2/api.go

```diff
@@ -42,44 +35,37 @@ var _ plugin.CreateAPI = &createAPIPlugin{}
 func (p *createAPIPlugin) UpdateContext(ctx *plugin.Context) { p.CreateAPI.UpdateContext(ctx) }
 func (p *createAPIPlugin) BindFlags(fs *pflag.FlagSet)       { p.CreateAPI.BindFlags(fs) }
 
+// InjectConfig will inject the PROJECT file/config in the plugin
 func (p *createAPIPlugin) InjectConfig(c *config.Config) {
 	p.CreateAPI.InjectConfig(c)
 	p.config = c
 }
 
+// Run will call the plugin actions
 func (p *createAPIPlugin) Run() error {
-	if err := p.CreateAPI.Run(); err != nil {
-		return err
-	}
+	return cmdutil.Run(p)
+}
 
+// Validate will perform the required checks before to start to scaffold
+func (p *createAPIPlugin) Validate() error {
 	// Emulate plugins phase 2 behavior by checking the config for this plugin's
 	// config object.
 	if !hasPluginConfig(p.config) {
 		return nil
 	}
-
-	return p.run()
+	return nil
 }
 
-// SDK plugin-specific scaffolds.
-func (p *createAPIPlugin) run() error {
-
-	// Write CR paths to the samples' kustomization file. This file has a
-	// "do not modify" comment so it is safe to overwrite.
-	samplesKustomization := sampleKustomizationFragment
-	for _, gvk := range p.config.Resources {
-		samplesKustomization += fmt.Sprintf("- %s\n", makeCRFileName(gvk))
-	}
-	kpath := filepath.Join("config", "samples")
-	if err := kustomize.Write(kpath, samplesKustomization); err != nil {
-		return err
+// Scaffold will run the plugin scaffold
+func (p *createAPIPlugin) GetScaffolder() (scaffold.Scaffolder, error) {
+	if err := p.CreateAPI.Run(); err != nil {
```

Comment:
It is confusing that `GetScaffolder` actually runs the scaffolding here, especially because this function does not execute scaffolding for helm.  https://github.com/operator-framework/operator-sdk/pull/3295/files#diff-38f829d64c0f3557998e226c42dd936dR172

Reply:
This should not be running the kubebuilder CreateAPI` plugin, this should be in `Run`.

Reply:
Hi @estroz and @asmacdo, 

Following some explanations.
 
`This should not be running the kubebuilder CreateAPIplugin, this should be inRun`.
We cannot keep it in the `Run`. See: 

- The run calls the interface to ensure that all plugins will follow the same workflow. 

```
// Run will call the plugin actions
func (p *createAPIPlugin) Run() error {
	return cmdutil.Run(p)
}
```

- Then, it will call : 

https://github.com/operator-framework/operator-sdk/blob/master/internal/kubebuilder/cmdutil/cmdutil.go#L22-L30

- See that the `GetScaffolder` must to return the Scaffold or nil (). 

Helm example:

```go
// GetScaffolder defines the templates that should be scaffold
func (p *createAPIPlugin) GetScaffolder() (scaffold.Scaffolder, error) {
	return scaffolds.NewAPIScaffolder(p.config, p.createOptions), nil
} 
```

- Then, the actions which are specific for the SDK CLI tool needs to be done in the `PostScaffold` since are not actually part of the Plugin Scaffold process and are done on top of that.

Reply:
The changes here were applying the plugin to follow up on the interface. But, I believe that I understand @estroz points as well. So, I reverted it and just keep the centralized code which shows address both concerns.